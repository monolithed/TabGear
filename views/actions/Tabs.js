import * as ActionTypes from '../constants/ActionTypes';

export const Tabs = {
	/**
	 * Show tabs
	 *
	 * @param {Array} tabs
	 * @returns {Object}
	 */
	showTabs (tabs) {
		return {
			type: ActionTypes.SHOW_TABS,
			api: true,
			tabs
		};
	},

	/**
	 * Disable tabs
	 *
	 * @param {Array} tabs
	 * @param {boolean} state
	 * @returns {Object}
	 */
	maskTabs (tabs, state) {
		return {
			type: ActionTypes.DISABLE_TABS,
			tabs,
			state
		};
	},

	/**
	 * Search tabs
	 *
	 * @param {Array} tabs
	 * @param {string} text
	 * @returns {Object}
	 */
	searchTabs (tabs, text) {
		return {
			type: ActionTypes.SEARCH_TABS,
			tabs,
			text
		};
	},

	/**
	 * Highlights the given tabs
	 *
	 * @param {string} index — the tab index to highlight
	 * @returns {Object}
	 */
	switchTab (index) {
		return {
			type: ActionTypes.SWITCH_TAB,
			api: true,
			index
		};
	},

	/**
	 * Closes selected tab
	 *
	 * @param {string} id — the tab id to close
	 * @returns {Object}
	 */
	closeTab (id) {
		return {
			type: ActionTypes.CLOSE_TAB,
			api: true,
			id
		};
	}
};

