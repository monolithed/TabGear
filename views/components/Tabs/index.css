@font-face {
	font-family: 'close';
	src: url('../../../files/fonts/close.woff?88812143') format('woff');
	font-weight: normal;
	font-style: normal;
}

:root {
	--item-padding: 15px;
	--item-outline: #d4d4d4;
}

.tg-tabs {
	border-radius: 3px;
	margin: 0 auto;
	list-style: none;
	padding: 0;
}

	.tg-tabs__item {
		border-bottom: 1px solid #e4e4e4;
		position: relative;
		transition: background-color .3s ease-in-out;
		width: 100%;
	}

		.tg-tabs__item-state_incognito {
			background-color: #efefef;
		}

		.tg-tabs__item-state_highlighted {
			background-color: #f2f6f9;
		}

		.tg-tabs__item-link {
			color: #999;
			position: relative;
			display: block;
			/*display: flex;*/
			/*align-items: flex-start;*/
			line-height: 18px;
			padding: var(--item-padding);
			padding-right: 20px;
			/*justify-content: flex-start;*/
			text-decoration: none;
			width: 100%;
		}

		.tg-tabs__item-link:focus {
			outline: 1px solid var(--item-outline);
			box-shadow: 1px 2px 12px var(--item-outline);
		}

			.tg-tabs__item-link:after,
			.tg-tabs__item-close {
				content: '\e800';
				font: normal 16px/16px close, sans-serif;
				opacity: 0;
				position: absolute;
				top: var(--item-padding);
				right: var(--item-padding);
				text-transform: none;
				-webkit-font-smoothing: antialiased;
				-moz-osx-font-smoothing: grayscale;
				transition: all 1s;
				z-index: 0;
			}

			.tg-tabs__item-link:hover:after,
			.tg-tabs__item-close:hover {
				opacity: 1;
			}

			.tg-tabs__item-close {
				height: 16px;
				width: 16px;
				z-index: 1;
			}

		.tg-tabs__item-icon {
			float: left;
			margin-right: var(--item-padding);
			height: 20px;
			width: 20px;
		}

		.tg-tabs__item-text {
			display: block;
			margin-right: 22px;
		}

	.tg-tabs__item:last-child {
		border-bottom: 0;
	}

	.tg-tabs__item:hover {
		background-color: #f5f5f5;
	}
