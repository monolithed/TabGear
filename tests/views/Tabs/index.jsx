import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow, mount, render } from 'enzyme';

import Tabs from '../../../views/components/Tabs';
import * as props from '../../../stubs/props';

test('Tabs view', () => {
	let states = ['', 'is-active', 'is-empty'];

	for (let state of states) {
		let component = renderer.create(<Tabs { ...props } state={ state } />);
		let view = component.toJSON();

		let actual = expect(view);
		actual.toMatchSnapshot();
	}
});
