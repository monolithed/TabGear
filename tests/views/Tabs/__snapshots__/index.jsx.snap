exports[`test Tabs view 1`] = `
<div>
  <ul
    className="tg-tabs">
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={0}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://cloud.google.com/_static/cc7442d8c7/images/cloud/icons/favicons/favicon.ico" />
        <span
          className="tg-tabs__item-text">
           
          Google Cloud Computing, Hosting Services & APIs | Google Cloud Platform
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={0}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={1}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://developers.google.com/site-assets/logo-google-plus-white.svg" />
        <span
          className="tg-tabs__item-text">
           
          Google Plus +++++++++++++++++++++++++++++++++++++++++++
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={1}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={2}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://developers.google.com/_static/cc7442d8c7/images/favicon.png" />
        <span
          className="tg-tabs__item-text">
           
          Google Play
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={2}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item tg-tabs__item--incognito">
      <a
        className="tg-tabs__item-link"
        data-index={3}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://www.google.com/images/branding/product/ico/maps_32dp.ico" />
        <span
          className="tg-tabs__item-text">
           
          Google Maps
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={3}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={4}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://translate.google.ru/favicon.ico" />
        <span
          className="tg-tabs__item-text">
           
          Google Translate
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={4}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={5}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://developer.android.com/favicon.ico" />
        <span
          className="tg-tabs__item-text">
           
          Android
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={5}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item tg-tabs__item--highlighted">
      <a
        className="tg-tabs__item-link"
        data-index={6}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://www.google.com/images/icons/product/chrome-32.png" />
        <span
          className="tg-tabs__item-text">
           
          Google Chrome
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={6}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={8}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://developers.google.com/web/images/web-fundamentals-icon192x192.png" />
        <span
          className="tg-tabs__item-text">
           
          Google Developers
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={8}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={8}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://firebase.google.com/_static/cc7442d8c7/images/firebase/favicon.png" />
        <span
          className="tg-tabs__item-text">
           
          Google Firebase
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={8}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
  </ul>
  <div>
    <div
      className="tg-body tg-body--disable is-inactive" />
    <div
      className="tg-tabs__search-results">
       
      <ul
        className="tg-tabs">
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={0}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://cloud.google.com/_static/cc7442d8c7/images/cloud/icons/favicons/favicon.ico" />
            <span
              className="tg-tabs__item-text">
               
              Google Cloud Computing, Hosting Services & APIs | Google Cloud Platform
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={0}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={1}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://developers.google.com/site-assets/logo-google-plus-white.svg" />
            <span
              className="tg-tabs__item-text">
               
              Google Plus +++++++++++++++++++++++++++++++++++++++++++
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={1}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={2}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://developers.google.com/_static/cc7442d8c7/images/favicon.png" />
            <span
              className="tg-tabs__item-text">
               
              Google Play
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={2}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item tg-tabs__item--incognito">
          <a
            className="tg-tabs__item-link"
            data-index={3}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://www.google.com/images/branding/product/ico/maps_32dp.ico" />
            <span
              className="tg-tabs__item-text">
               
              Google Maps
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={3}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={4}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://translate.google.ru/favicon.ico" />
            <span
              className="tg-tabs__item-text">
               
              Google Translate
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={4}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={5}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://developer.android.com/favicon.ico" />
            <span
              className="tg-tabs__item-text">
               
              Android
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={5}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item tg-tabs__item--highlighted">
          <a
            className="tg-tabs__item-link"
            data-index={6}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://www.google.com/images/icons/product/chrome-32.png" />
            <span
              className="tg-tabs__item-text">
               
              Google Chrome
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={6}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={8}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://developers.google.com/web/images/web-fundamentals-icon192x192.png" />
            <span
              className="tg-tabs__item-text">
               
              Google Developers
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={8}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={8}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://firebase.google.com/_static/cc7442d8c7/images/firebase/favicon.png" />
            <span
              className="tg-tabs__item-text">
               
              Google Firebase
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={8}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
      </ul>
       
    </div>
  </div>
</div>
`;

exports[`test Tabs view 2`] = `
<div>
  <ul
    className="tg-tabs is-active">
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={0}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://cloud.google.com/_static/cc7442d8c7/images/cloud/icons/favicons/favicon.ico" />
        <span
          className="tg-tabs__item-text">
           
          Google Cloud Computing, Hosting Services & APIs | Google Cloud Platform
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={0}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={1}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://developers.google.com/site-assets/logo-google-plus-white.svg" />
        <span
          className="tg-tabs__item-text">
           
          Google Plus +++++++++++++++++++++++++++++++++++++++++++
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={1}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={2}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://developers.google.com/_static/cc7442d8c7/images/favicon.png" />
        <span
          className="tg-tabs__item-text">
           
          Google Play
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={2}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item tg-tabs__item--incognito">
      <a
        className="tg-tabs__item-link"
        data-index={3}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://www.google.com/images/branding/product/ico/maps_32dp.ico" />
        <span
          className="tg-tabs__item-text">
           
          Google Maps
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={3}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={4}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://translate.google.ru/favicon.ico" />
        <span
          className="tg-tabs__item-text">
           
          Google Translate
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={4}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={5}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://developer.android.com/favicon.ico" />
        <span
          className="tg-tabs__item-text">
           
          Android
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={5}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item tg-tabs__item--highlighted">
      <a
        className="tg-tabs__item-link"
        data-index={6}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://www.google.com/images/icons/product/chrome-32.png" />
        <span
          className="tg-tabs__item-text">
           
          Google Chrome
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={6}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={8}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://developers.google.com/web/images/web-fundamentals-icon192x192.png" />
        <span
          className="tg-tabs__item-text">
           
          Google Developers
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={8}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={8}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://firebase.google.com/_static/cc7442d8c7/images/firebase/favicon.png" />
        <span
          className="tg-tabs__item-text">
           
          Google Firebase
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={8}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
  </ul>
  <div>
    <div
      className="tg-body tg-body--disable is-active" />
    <div
      className="tg-tabs__search-results">
       
      <ul
        className="tg-tabs is-active">
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={0}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://cloud.google.com/_static/cc7442d8c7/images/cloud/icons/favicons/favicon.ico" />
            <span
              className="tg-tabs__item-text">
               
              Google Cloud Computing, Hosting Services & APIs | Google Cloud Platform
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={0}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={1}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://developers.google.com/site-assets/logo-google-plus-white.svg" />
            <span
              className="tg-tabs__item-text">
               
              Google Plus +++++++++++++++++++++++++++++++++++++++++++
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={1}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={2}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://developers.google.com/_static/cc7442d8c7/images/favicon.png" />
            <span
              className="tg-tabs__item-text">
               
              Google Play
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={2}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item tg-tabs__item--incognito">
          <a
            className="tg-tabs__item-link"
            data-index={3}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://www.google.com/images/branding/product/ico/maps_32dp.ico" />
            <span
              className="tg-tabs__item-text">
               
              Google Maps
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={3}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={4}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://translate.google.ru/favicon.ico" />
            <span
              className="tg-tabs__item-text">
               
              Google Translate
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={4}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={5}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://developer.android.com/favicon.ico" />
            <span
              className="tg-tabs__item-text">
               
              Android
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={5}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item tg-tabs__item--highlighted">
          <a
            className="tg-tabs__item-link"
            data-index={6}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://www.google.com/images/icons/product/chrome-32.png" />
            <span
              className="tg-tabs__item-text">
               
              Google Chrome
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={6}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={8}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://developers.google.com/web/images/web-fundamentals-icon192x192.png" />
            <span
              className="tg-tabs__item-text">
               
              Google Developers
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={8}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={8}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://firebase.google.com/_static/cc7442d8c7/images/firebase/favicon.png" />
            <span
              className="tg-tabs__item-text">
               
              Google Firebase
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={8}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
      </ul>
       
    </div>
  </div>
</div>
`;

exports[`test Tabs view 3`] = `
<div>
  <ul
    className="tg-tabs is-empty">
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={0}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://cloud.google.com/_static/cc7442d8c7/images/cloud/icons/favicons/favicon.ico" />
        <span
          className="tg-tabs__item-text">
           
          Google Cloud Computing, Hosting Services & APIs | Google Cloud Platform
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={0}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={1}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://developers.google.com/site-assets/logo-google-plus-white.svg" />
        <span
          className="tg-tabs__item-text">
           
          Google Plus +++++++++++++++++++++++++++++++++++++++++++
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={1}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={2}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://developers.google.com/_static/cc7442d8c7/images/favicon.png" />
        <span
          className="tg-tabs__item-text">
           
          Google Play
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={2}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item tg-tabs__item--incognito">
      <a
        className="tg-tabs__item-link"
        data-index={3}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://www.google.com/images/branding/product/ico/maps_32dp.ico" />
        <span
          className="tg-tabs__item-text">
           
          Google Maps
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={3}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={4}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://translate.google.ru/favicon.ico" />
        <span
          className="tg-tabs__item-text">
           
          Google Translate
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={4}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={5}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://developer.android.com/favicon.ico" />
        <span
          className="tg-tabs__item-text">
           
          Android
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={5}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item tg-tabs__item--highlighted">
      <a
        className="tg-tabs__item-link"
        data-index={6}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://www.google.com/images/icons/product/chrome-32.png" />
        <span
          className="tg-tabs__item-text">
           
          Google Chrome
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={6}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={8}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://developers.google.com/web/images/web-fundamentals-icon192x192.png" />
        <span
          className="tg-tabs__item-text">
           
          Google Developers
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={8}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
    <li
      className="tg-tabs__item">
      <a
        className="tg-tabs__item-link"
        data-index={8}
        data-name="tab"
        href="#"
        onClick={[Function]}>
        <img
          className="tg-tabs__item-icon"
          src="https://firebase.google.com/_static/cc7442d8c7/images/firebase/favicon.png" />
        <span
          className="tg-tabs__item-text">
           
          Google Firebase
           
        </span>
        <span
          className="tg-tabs__item-close tg-icon"
          data-id={8}
          onClick={[Function]}
          title="Close" />
      </a>
    </li>
  </ul>
  <div>
    <div
      className="tg-body tg-body--disable is-active" />
    <div
      className="tg-tabs__search-results">
       
      <ul
        className="tg-tabs is-empty">
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={0}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://cloud.google.com/_static/cc7442d8c7/images/cloud/icons/favicons/favicon.ico" />
            <span
              className="tg-tabs__item-text">
               
              Google Cloud Computing, Hosting Services & APIs | Google Cloud Platform
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={0}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={1}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://developers.google.com/site-assets/logo-google-plus-white.svg" />
            <span
              className="tg-tabs__item-text">
               
              Google Plus +++++++++++++++++++++++++++++++++++++++++++
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={1}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={2}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://developers.google.com/_static/cc7442d8c7/images/favicon.png" />
            <span
              className="tg-tabs__item-text">
               
              Google Play
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={2}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item tg-tabs__item--incognito">
          <a
            className="tg-tabs__item-link"
            data-index={3}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://www.google.com/images/branding/product/ico/maps_32dp.ico" />
            <span
              className="tg-tabs__item-text">
               
              Google Maps
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={3}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={4}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://translate.google.ru/favicon.ico" />
            <span
              className="tg-tabs__item-text">
               
              Google Translate
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={4}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={5}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://developer.android.com/favicon.ico" />
            <span
              className="tg-tabs__item-text">
               
              Android
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={5}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item tg-tabs__item--highlighted">
          <a
            className="tg-tabs__item-link"
            data-index={6}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://www.google.com/images/icons/product/chrome-32.png" />
            <span
              className="tg-tabs__item-text">
               
              Google Chrome
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={6}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={8}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://developers.google.com/web/images/web-fundamentals-icon192x192.png" />
            <span
              className="tg-tabs__item-text">
               
              Google Developers
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={8}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
        <li
          className="tg-tabs__item">
          <a
            className="tg-tabs__item-link"
            data-index={8}
            data-name="tab"
            href="#"
            onClick={[Function]}>
            <img
              className="tg-tabs__item-icon"
              src="https://firebase.google.com/_static/cc7442d8c7/images/firebase/favicon.png" />
            <span
              className="tg-tabs__item-text">
               
              Google Firebase
               
            </span>
            <span
              className="tg-tabs__item-close tg-icon"
              data-id={8}
              onClick={[Function]}
              title="Close" />
          </a>
        </li>
      </ul>
       
    </div>
  </div>
</div>
`;
